Common/cpp/LayoutAnimations/LayoutAnimationsManager.cpp

```diff
@@ -73,4 +91,18 @@ void LayoutAnimationsManager::startLayoutAnimation(
       config->getJSValue(rt));
 }
 
+int LayoutAnimationsManager::findSiblingForSharedView(int tag) {
+  /*
+    The top screen on the stack triggers the animation, so we need to find
+    the sibling view registered in the past. That's why we look backward.
+   */
+  for (auto const &[key, group] : sharedTransitionGroups_) {
+    auto position = std::find(group.begin(), group.end(), tag);
+    if (position != group.end() && position != group.begin()) {
+      return *std::prev(position);
+    }
```

Comment:
1. I don't fully understand what this function is supposed to do.
2. What about `key`? Is it needed?
3. It looks like this can be done easier

Reply:
I'd suggest to call it differently. Also typically it's better to add method-level documentation prior to the definition not within the method.

I think that "sibling" doesn't capture the meaning of what we are searching here in this method. I'd call it "findPrecedingViewTagForTransition" and describe that it looks for a veiw tegistered in the same transition group (that is with the same tag) which has been added to that group directly before the one that we provide as an argument.

Reply:
I agree with @tomekzaw here, I think it would be better to have a map `tag -> sharedTransitionTag` and use it to find the shared transition group for the tag, instead of looping.

Also, the naming here isn't perfect as @kmagiera noted.